### PR TITLE
[SPARK-19373][MESOS] Base spark.scheduler.minRegisteredResourceRatio on registered cores rather than accepted cores

### PR DESCRIPTION
**Pull request by mgummelt**
## What changes were proposed in this pull request?

See JIRA

## How was this patch tested?

Unit tests, Mesos/Spark integration tests

cc @skonto @susanxhuynh

### CODE REVIEW

File: resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala [hunk 1]
```diff
@@ -256,7 +259,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   }
 
   override def sufficientResourcesRegistered(): Boolean = {
-    totalCoresAcquired >= maxCores * minRegisteredRatio
+    totalCoreCount.get >= maxCoresOption.getOrElse(0) * minRegisteredRatio
```
Comment (mgummelt) [inline resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala]: This is the only substantive change.

**Comment by markhamstra** [comment]
Please avoid using "fix" as the description in a PR -- it doesn't tell us anything substantive about the nature of the problem or its resolution, so any future reviewing of commit messages will require digging into the actual committed code to get even a minimal idea of what was done.

**Comment by mgummelt** [comment]
@markhamstra Updated

**Comment by markhamstra** [comment]
thanks

File: resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala (hunk 1 above)
Comment (skonto) [inline resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala]: This means return fast most of the time. `totalCoreCount` holds the total number of cores in the cluster. With a reasonable max core value this inequality will usually hold, right? This means immediate task schedule? In the Jira the opposite is described, am I missing something?

**Comment by skonto** [comment]
LGTM run it locally works fine.

File: resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala (hunk 1 above)
Comment (mgummelt) [inline resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala]: Yep.

**Comment by mgummelt** [comment]
@srowen Can we get a merge?  This is a bugfix, so it probably belongs in all supported branches (1.6, 2.0, 2.1, master)

**Comment by srowen** [comment]
Merged to master. The cherry-pick wasn't clean into 2.1, and it wasn't entirely trivial. Rather than try to fix and get it wrong, if you'd care to open a PR against 2.1 I can merge that back and see how far that gets.

**Pull request closed without merging**